fe/fe-core/src/main/java/com/starrocks/qe/HDFSBackendSelector.java

```diff
@@ -171,6 +171,15 @@ private ComputeNode reBalanceScanRangeForComputeNode(List<ComputeNode> backends,
             return null;
         }
 
+        boolean forceReBalance = ConnectContext.get() != null ? ConnectContext.get().getSessionVariable().
+                getHdfsBackendSelectorForceRebalance() : false;
+        boolean enableDataCache = ConnectContext.get() != null ? ConnectContext.get().getSessionVariable().
+                isEnableScanDataCache() : false;
+        // If force-rebalancing is not specified and cache is used, skip the rebalancing directly.
+        if (!forceReBalance && enableDataCache) {
```

Comment:
maybe we need add user guide that only when cache enabled, forceReBalance does it's work.

Reply:
> maybe we need add user guide that only when cache enabled, forceReBalance does it's work.

Ok，set it invisible currently, and if it is necessary to expose it to users, we will add relevant documentation to explain it